hypershift-operator/controllers/hostedclustersizing/hostedclustersizing_controller.go

```diff
@@ -232,7 +235,25 @@ func (r *reconciler) reconcile(
 		}
 
 		for _, nodePool := range nodePools.Items {
-			nodeCount += uint32(nodePool.Status.Replicas)
+			var replicas uint32
+			// If autoscaling, the replicas should be returned from status
+			if nodePool.Spec.AutoScaling != nil {
+				nodeCountRequiresAPIServer = true
```

Comment:
Alternatively to use `nodeCountRequiresAPIServer = true` here and in l222

We could also fetch the condition first and return early when needed. E.g.
Then in both places we'd do
```
if !kasAvailable && replicas == 0 { 
  log
  return nil, nil
}
```
Just an alternative thought, feel free to ignore it.